### ISSUE TITLE
Add ARMv6 support

### ISSUE THREAD

**Issue by liyuqian** [flutter/flutter#22380]
According to https://github.com/flutter/engine/wiki/Android-Builds-Supporting-armeabi, Alibaba needs that support because Taobao of Alibaba and QQ of Tencent both use ARMv6 to support as many devices as possible.

We should be able to modify our GN files to support it without much work according to the Wiki above.

**Comment by galvesribeiro** [comment]
Helllo folks!

According to that wiki, it is focused on Android based devices (hence the requirements for NDK). As I asked on #22492, I would like to run flutter on generic Linux ARMv6l without any X (directly using Linux framebuffer /dev/fb) using Skia and software rendering. 

Can someone do the same and put together a wiki page for that? 

Thanks!

**Comment by mit-mit** [comment]
As per the Flutter FAQ, the only supported ARM platforms are ARMv7 and ARM64:
https://flutter.io/faq/#what-devices-and-os-versions-does-flutter-run-on

**Comment by galvesribeiro** [comment]
@mit-mit I got it. By default Skia is supposed to work on ARMv7 but we can make it work on v6 (already tested it) after messing with several GN files and disabling 3d acceleration in favor of software rendering. Is that the same reason Flutter is v7-only or is there other dependencies that requires v7 instruction set?

**Comment by mit-mit** [comment]
Yes, ARMv6 support also requires support in Dart AOT compilation. We already support a lot of different variations in the AOT compiler, and it adds a lot of complexity to have many of them.

**Comment by jwinarske** [comment]
@mit-mit [Dart SDK for linux-armv6l](https://chrome-infra-packages.appspot.com/p/dart/dart-sdk/linux-armv6l/+/) looks active.  Something isn't adding up.

**Comment by athomas** [comment]
@jwinarske 
**_TL;DR: This is a differently labelled regular ARMv7 SDK because of a quirk in Luci's CIPD packaging tool._**

[This](https://chrome-infra-packages.appspot.com/p/dart/dart-sdk/linux-armv6l/+/) is the same thing we have [on the website labelled as ARMv7](https://dart.dev/tools/sdk/archive).

The script that uploads it [is hardcoded to ARMv6l](https://github.com/dart-lang/sdk/blob/7929b4223e7cde725126b3177b6e8f95855bbcea/tools/sdks/update.sh#L37).

[The reason](https://chromium.googlesource.com/infra/luci/luci-go/+/8dfb4be10dd0c7e71a74acc0bb2e184785e212ce/cipd/client/cipd/platform/platform.go#30) for that is that CIPD identifies all ARM (32bit) architectures as `armv6l` (`GOARCH` would be linux/arm here).

The arm SDK releases on dart.dev are all built for ARMv7. We have no ARMv6 compatible distributions at the moment.

**Comment by jwinarske** [comment]
@athomas Ah!  Okay.  I thought I was onto something.  Thanks for confirming.

**Comment by mit-mit** [comment]
Breakign change request to completely remove ARMv6 support: https://github.com/dart-lang/sdk/issues/42069

**Comment by mit-mit** [comment]
Closing the present issue; we are moving forward with **removing ARMv6 support**. For any follow-up discussion, please use https://github.com/dart-lang/sdk/issues/42069
